File: WebViewApp/app/src/main/java/com/example/webviewapp/contract/HistoryContract.java
package com.example.webviewapp.contract;

import com.example.webviewapp.data.Record;

import java.util.List;

public interface HistoryContract {
    interface View {
        void setEditTextVisibility(Boolean isVisible);
    }

    interface Presenter {
        List<Record> getData();

        List<Record> refreshRecord();

        void checkScrolled(int dy);
    }
}

File: WebViewApp/app/src/main/java/com/example/webviewapp/contract/HistoryContract.kt
package com.example.webviewapp.contract

import com.example.webviewapp.data.Record

interface HistoryContract {
    interface View {
        fun setEditTextVisibility(isVisible: Boolean?)
    }

    interface Presenter {
        val data: List<Record?>?
        fun refreshRecord(): List<Record?>?
        fun checkScrolled(dy: Int)
        fun deleteAllHistory()
        fun deleteHistoryByUrl(url: List<String?>?)
    }
}